src/container/container.ts

```diff
@@ -408,7 +408,7 @@ class Container implements interfaces.Container {
     if (Reflect.hasMetadata(METADATA_KEY.PRE_DESTROY, constructor)) {
       const data: interfaces.Metadata = Reflect.getMetadata(METADATA_KEY.PRE_DESTROY, constructor);
 
-      return instance[data.value]();
+      return instance[data.value as number]();
```

Comment:
again why the cast to number ?

Reply:
Type 'unknown' cannot be used as an index type.ts(2538)